include/openssl/rsa.h

```diff
@@ -859,5 +859,6 @@ BSSL_NAMESPACE_END
 #define RSA_R_PUBLIC_KEY_VALIDATION_FAILED 146
 #define RSA_R_D_OUT_OF_RANGE 147
 #define RSA_R_BLOCK_TYPE_IS_NOT_02 148
+#define RSA_R_MISMATCHED_SIGNATURE 149
```

Comment:
Maybe we should reserve some space for openssl/boringssl defined values here. Offset of 100 reasonable?

Reply:
yep, sounds good.

> Feel free to add more tests - if you want to play around with it and can come up with some. But not a blocker.

i tried to identify new cases relevant to this change, but it looks like the existing test cases cover the "well-formed, but mismatched" cases pretty well.